test/unit/math/gpu/copy_triangular_test.cpp

```diff
@@ -0,0 +1,44 @@
+#ifdef STAN_OPENCL
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/gpu/copy_triangular.hpp>
+#include <gtest/gtest.h>
+#include <algorithm>
+
+TEST(MathMatrixGPU, copy_triangular_m_exception_pass) {
+  stan::math::matrix_gpu m0;
+
+  EXPECT_NO_THROW(stan::math::copy_triangular<stan::math::Upper>(m0));
+  EXPECT_NO_THROW(stan::math::copy_triangular<stan::math::Lower>(m0));
+
+  stan::math::matrix_gpu m1(1, 1);
+
+  EXPECT_NO_THROW(stan::math::copy_triangular<stan::math::Upper>(m1));
+  EXPECT_NO_THROW(stan::math::copy_triangular<stan::math::Lower>(m1));
+}
+
```

Comment:
Can we test the behavior of this triangular copy when it's not square?

Reply:
I added 2 matrices with different sizes (1 with more rows, 1 with more cols) and two tests for both cases.